scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -521,11 +521,11 @@ trait CounterVerification extends Job {
   }
 }
 
-private[scalding] object FlowStepStrategies {
+private[scalding] case class FlowStepStrategies[A]() extends Semigroup[FlowStepStrategy[A]] {
```

Comment:
Should this be a Semigroup? Looks like a side-effecting function here. Are we guaranteed associativity?